cache/test_cache.py

```diff
@@ -94,4 +95,72 @@ class TestCacheReplicated(TestCacheDisabled):
     # Replicated mode.
     cache_mode = 2
 
-# vim: tabstop=8 expandtab shiftwidth=4 softtabstop=4
+class H2Cache(tester.TempestaTest):
+    clients = [{
+        'id': 'deproxy',
+        'type': 'deproxy_h2',
+        'addr': "${server_ip}",
+        'port': '443',
+        'ssl': True,
+        'ssl_hostname': 'tempesta-tech.com'
+    }]
+
+    backends = [{
+        'id' : 'deproxy',
+        'type' : 'deproxy',
+        'port' : '8000',
+        'response' : 'static',
+        'response_content' :
+        'HTTP/1.1 200 OK\r\n'
+        'Content-Length: 0\r\n\r\n'
+    }]
+
+    tempesta = {
+        'config':
+        '''
+        cache 2;
+        cache_fulfill eq /to-be-cached;
+
+        listen 443 proto=h2;
+        tls_match_any_server_name;
+
+        srv_group default {
+            server ${server_ip}:8000;
+        }
+
+        vhost tempesta-tech.com {
+            tls_certificate ${tempesta_workdir}/tempesta.crt;
+            tls_certificate_key ${tempesta_workdir}/tempesta.key;
+            proxy_pass default;
+        }
+
+       '''
+    }
+
+    def start_all(self):
+        self.start_all_servers()
+        self.start_tempesta()
+        self.deproxy_manager.start()
+        self.start_all_clients()
+        self.assertTrue(self.wait_all_connections())
+
+    def test(self):
+        self.start_all()
+
+        request = [
+            (":authority", "tempesta-tech.com"),
+            (":path", "/to-be-cached"),
+            (":scheme", "https"),
+            (":method", "GET")
+        ]
+        requests = [request, request]
+
+        client = self.get_client("deproxy")
+        client.make_requests(requests)
+
+        got_response = client.wait_for_response(timeout=5)
+
+        print(f"Amount of requests passed to the backend: {len(self.get_server('deproxy').requests)}")
```

Comment:
It seems here should be an assertion, that backend receives only 1 request. Also too wide line